appshell/appshell_extensions_mac.mm

```
@@ -1175,6 +1175,154 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
+            
+            args.push_back(rmArgs);
+            args.push_back(destFile);
+            
+            toolStatus = _RunToolWithAdminPrivileges(authorizationRef, rmTool, args);
+            
+            if( toolStatus == errAuthorizationSuccess) {
+                
+                // *** Check and create if the directory is not present ***
+                
+                args.clear();
+                
+                args.push_back(mkDirArgs);
+                args.push_back(destFolder);
+                
+                toolStatus = _RunToolWithAdminPrivileges(authorizationRef, mkDirTool, args);
+                
+                if( toolStatus == errAuthorizationSuccess) {
+                    
+                    // *** Go ahead and create the symlink now ***
+                    
+                    args.clear();
+                    
+                    args.push_back(lnArgs);
+                    args.push_back(sourceFile);
+                    args.push_back(destFile);
+                    
+                    toolStatus = _RunToolWithAdminPrivileges(authorizationRef, lnTool, args);
+
+                    if( toolStatus != errAuthorizationSuccess)
+                        errorCode = ERR_CL_TOOLS_SYMLINKFAILED;
+                    
+                }
+                else {
+                    errorCode = ERR_CL_TOOLS_MKDIRFAILED;
+                }
+            }
+            else{
+                if(toolStatus == errAuthorizationCanceled)
+                    errorCode = ERR_CL_TOOLS_CANCELLED;
+                else
+                    errorCode = ERR_CL_TOOLS_RMFAILED;
+            }
+
+        } catch (...) {
+            // This is empty as the below statements will take care of
+            // releasing authorizationRef.
+            errorCode = ERR_CL_TOOLS_SERVFAILED;
+        }
+        
+        if(authorizationRef) {
+            AuthorizationFree(authorizationRef, kAuthorizationFlagDestroyRights);
```

This call could also return an OSStatus equivalent of error. Could we return an error here, something like 'AUTH_RELEASE_FAILED'? We could bail on that error. Is there a possibility of some other extension re-using this authref